src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -50,6 +50,20 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         set;
     }
 
+    /***
+    * @description Validate Recurring Donations Data.
+    */
+    @TestVisible
+    private RD2_ValidationService validationService {
```

Comment:
Since `validationService` is not set from tests, it can be declared and defined in the `validate(List<SObject> rds, TDTM_Runnable.Action triggerAction)`.

Reply:
@RadSFDeveloper , Do you suggest to remove this initialization of validationservice?

Reply:
The validation service can be initialized in the `validate()` method:
```
public void validate(List<SObject> rds, TDTM_Runnable.Action triggerAction) {

    if (triggerAction == TDTM_Runnable.Action.BeforeInsert 
        || triggerAction == TDTM_Runnable.Action.BeforeUpdate
    ) {
        new RD2_ValidationService().validate(rds);
    }
}
```